client/src/Components/FilterSection/index.tsx

```diff
@@ -2,51 +2,120 @@ import React, { useContext, useState, useEffect } from 'react';
 import { useLocation } from 'react-router-dom';
 import { Input, Select } from 'antd';
 import { useTranslation } from 'react-i18next';
-import './style.css';
-import { ServiceLocation } from '../../Context/ServiceLocationContext';
-import { locationObject, serviceObject, request } from '../../utils';
 import ErrorComponent from '../Error';
+import { ServiceLocation } from '../../Context/ServiceLocationContext';
+import './style.css';
+import {
+  locationObject,
+  serviceObject,
+  request,
+  FilterSearchProps,
+  SearchResponse,
+  Event,
+} from '../../utils';
 
-function FilterSection() {
+function FilterSection(
+  {
+    setUsers,
+    setSearchError,
+    setIsLoading,
+    setResultCount,
+    setPage,
+    page,
+  }: FilterSearchProps,
+) {
   const { t } = useTranslation();
   const locationData = useLocation();
-  const search : any = locationData.state;
-  const { serviceSearch, locationSearch } = search;
-
   const [mainServiceId, setMainServiceId] = useState(0);
   const [subService, setSubService] = useState<number[]>([]);
   const [error, setError] = useState('');
   const [subServices, setSubServices] = useState<serviceObject[]>([]);
-
+  const [name, setName] = useState<string>('');
+  const [locationInput, setLocationInput] = useState<number>();
+  const search : any = locationData.state;
+  const { serviceSearch, locationSearch } = search;
   const { data: { location, services } } = useContext(ServiceLocation);
   const { Option } = Select;
-
-  useEffect(() => {
-    const getSubServicesData = async () => {
-      try {
-        setError('');
-        const { data } : any = await request('get', `/subservices/${+mainServiceId || serviceSearch}`);
-        setSubServices(data);
-      } catch (responseError: any) {
-        setError(responseError.data.msg);
-      }
-    };
-    if (mainServiceId || serviceSearch) getSubServicesData();
-  }, [mainServiceId, serviceSearch]);
-
+  const getSearchResults = async (queryString: string) => {
+    try {
+      setSearchError('');
+      const searchResult : SearchResponse = await request('get', `/provider?${queryString}`);
+      setIsLoading(false);
+      setResultCount(searchResult.count);
+      setUsers(searchResult.data);
```

Comment:
What if we declare the function outside the component and pass to it a success and failure callback
```js
  const getSearchResults = async (queryString: string, successCB, faluireCB) => {};
  ```